3rdparty/fastcv/src/fastcv_hal_core.cpp

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+*/
+
+#include "fastcv_hal_core.hpp"
+#include "fastcv_hal_utils.hpp"
+#include <opencv2/core/core.hpp>
+#include <opencv2/core/base.hpp>
+#include <opencv2/core/utils/logger.hpp>
+
+bool isInitialized = false;
```

Comment:
C++ does not define globals initialization order. In case of static linkage some OpenCV function may be called before the initialization here e.g. from constructor of global object. The state of the flag will be undefined here.

Reply:
will replace with singleton pattern